### PR TITLE
earlyoom: Apply security recommendations from package maintainer (First attempt)

### PR DESCRIPTION
**Pull request by kaindlnetwork**
My pull request might not be the most elegant solution, but it implements the security restrictions recommended by the package maintainer. I would appreciate any help with verifying and testing the changes. Thank you very much! :D

Source for my changes: https://github.com/rfjakob/earlyoom/blob/master/earlyoom.service.in

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Hello. Firstly, welcome to nixpkgs.

Your changes aren't indented correctly, if you could address that it'd be great.
Please also view CONTRIBUTING.md to see how commits should be titled.

If you can link the documentation from easyoom which recommends these options that would be helpful for review.

**Comment by kaindlnetwork** [comment]
Hi @06kellyjac,

Thank you so much for your suggestions!
I've updated my Pull Request to include the documentation I referenced for my change. Additionally, I’ve edited the title of the Pull Request to better reflect the changes.

Could you please let me know what is incorrect or missing to make my commit approvable? Specifically, regarding the comment "Your changes aren't indented correctly."

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
By indentation I mean this

image

It was originally indented twice (2 spaces each indentation) by a total of 4 spaces. Now it's not indented.

Also your commits will need squashing together which I don't think you can do via the github interface

File: nixos/modules/services/system/earlyoom.nix [hunk 1]
```diff
@@ -149,25 +149,52 @@ in
...
+       + optionalString (cfg.freeSwapKillThreshold != null) ",${toString cfg.freeSwapKillThreshold}")
+      "-r ${toString cfg.reportInterval}"
+    ]
+    ++ optionals cfg.enableDebugInfo [ "-d" ]
+    ++ optionals cfg.enableNotifications [ "-n" ]
+    ++ optionals (cfg.killHook != null) [ "-N ${escapeShellArg cfg.killHook}" ]
+    ++ cfg.extraArgs);
+
+    AmbientCapabilities = "CAP_KILL CAP_IPC_LOCK";
+    CapabilityBoundingSet = "CAP_KILL CAP_IPC_LOCK";
```
[the remark this answers is not in the document]
Comment (kaindlnetwork) [reply nixos/modules/services/system/earlyoom.nix]: I have applied your suggestion directly :D

File: nixos/modules/services/system/earlyoom.nix [hunk 2]
```diff
@@ -149,25 +149,52 @@ in
...
+    ProtectHostname = "true";
+    ProtectKernelLogs = "true";
+    ProtectKernelModules = "true";
+    ProtectKernelTunables = "true";
+    ProtectControlGroups = "true";
+    RestrictNamespaces = "true";
+    RestrictRealtime = "true";
+    LockPersonality = "true";
+    PrivateNetwork = "true";
+    IPAddressDeny = "true";
```
[the remark this answers is not in the document]
Comment (kaindlnetwork) [reply nixos/modules/services/system/earlyoom.nix]: I have applied your suggestion directly :D

File: nixos/modules/services/system/earlyoom.nix [hunk 3]
```diff
@@ -149,25 +149,52 @@ in
...
+    RestrictAddressFamilies = "AF_UNIX";
+    SystemCallArchitectures = "native";
+    SystemCallFilter = ["@system-service" "~@resources @privileged"];
```
[the remark this answers is not in the document]
Comment (kaindlnetwork) [reply nixos/modules/services/system/earlyoom.nix:201]: `    SystemCallFilter = ["@system-service" "~@resources @privileged"];` This is in line 47-48
`    RestrictAddressFamilies = "AF_UNIX"; ` This is in line 44
`    SystemCallArchitectures = "native"; ` This is in line 46
`IPAddressDeny=any` This is in line 41

I would like to keep these lines if it is okay :)

**Comment by kaindlnetwork** [comment]
Hi @06kellyjac,  

I've implemented all your suggestions except for the `SystemCallFilter` so far. I must say, NixOS has impressively high standards for PRs!  

To be honest, I'm not sure how to squash the commits. If you could assist with that, I’d really appreciate it (I rarely use Git commands). It looks like all the checks have passed successfully so far. :D  

Thanks!

**Pull request closed without merging**